sirepo/job_api.py

```diff
@@ -134,6 +135,8 @@ def api_runSimulation():
     # TODO(e-carlin): This should really be done in job_supervisor._lib_dir_symlink()
     # but that is outside of the Flask context so it won't work
     r.simulation_lib_dir = sirepo.simulation_db.simulation_lib_dir(r.simulationType)
+    if r.isParallel:
+        r.isPremiumUser = sirepo.auth.is_premium_user(r.uid)
```

Comment:
I don't think it is correct to denormalize r.uid here. Rather, is_premium_user should always be on the auth_user. This avoids problems if there's a bug related to setting r.uid.

Reply:
I think I understand the security implications but I'm not sure I understand "is_premium_user should always be on the auth_user". Should I not pass uid into is_premium_user and the function should get the current user (`logged_in_user()`)?

Reply:
Per our conversation: yes, you should rely on logged_in_user. auth_user is a BOP-concept. Sorry for the confusion.